src/ngraph/pass/convert_elimination.cpp

```diff
@@ -0,0 +1,91 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <functional>
+#include <memory>
+#include <typeindex>
+#include <typeinfo>
+#include <unordered_map>
+
+#include "ngraph/graph_util.hpp"
+#include "ngraph/log.hpp"
+#include "ngraph/op/convert.hpp"
+#include "ngraph/opsets/opset3.hpp"
+#include "ngraph/pass/convert_elimination.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+//************************
+// Eliminates 2 types of spines with converts:
+//
+// 1. [ op1--> convert2(type i64->i32)--> ReduceMin--> convert1(type i32->i64)--> op2 ] to
+//                    [ op1--> ReduceMin--> op2 ]
+// 2. [ op3--> convert1-->  NonZero--> op4 ] to
+//                    [ op3--> NonZero--> op4 ]
+//
+//*************************
+bool pass::ConvertElimination::eliminate_convert(const std::shared_ptr<Node>& node)
+{
+    auto convert = as_type_ptr<opset3::Convert>(node);
+    auto destination_type = convert->get_destination_type();
+    auto input_op = convert->input_value(0);
+
+    // case 1
+    if (is_type<opset3::ReduceMin>(input_op.get_node()))
+    {
+        auto convert_2 = input_op.get_node()->input_value(0).get_node();
+        if (is_type<opset3::Convert>(convert_2) && convert_2->get_users().size() == 1 &&
+            destination_type == convert_2->get_input_element_type(0))
+        {
+            if (destination_type == element::i64 && convert_2->get_element_type() == element::i32)
+            {
+                // replace ReplaceMin as convert's output
+                if (replace_output_update_name(convert->output(0), input_op))
```

Comment:
`replace_output_update_name` can be used only in case when you eliminate operation. But here you replace sub-graph convert-reduce-convert with reduce op. 
Please use two different patterns for detecting convert-reduce-convert and convert-nonzero. And when you replace subgraph with node use replace_node() and do not forget about copy_rt_info() and friendly names.

Reply:
Not sure what you mean by `Please use two different patterns for detecting convert-reduce-convert and convert-nonzero.` We already have an if - else enclosure.